### PR TITLE
boards: ct: ctcc: add ctcc/nrf52840 board definition

### PR DESCRIPTION
**Pull request by kowalewskijan**
Adds CTHINGS.CO Connectivity Card nRF52840 (ctcc_nrf52840) board support.

Tested samples using mPCIe - USB adapter:
- subsys/usb/console
- subsys/usb/dfu
- bluetooth/hci_usb

Additional tests:
- tested DFU update over USB with MCUboot and DFU Zephyr application

### CODE REVIEW

**Comment by kowalewskijan** [comment]
@galak seems like the CI is failing to unrelated problem to my PR, other than that all jobs are green.

**Review by decsny — DISMISSED** [review]

**Comment by kowalewskijan** [comment]
Hi @galak, friendly bump, can we merge this?

**Review by cvinayak — COMMENTED** [review]
Don't we need a sample to cover this board in CI? Also, should  a maintainer be assigned for this board?

File: boards/ct/ctcc_nrf52840/board.yml
Comment (fabiobaltieri) [inline boards/ct/ctcc_nrf52840/board.yml:8]: I don't think you need this if there's only one revision out

File: boards/ct/ctcc_nrf52840/ctcc_nrf52840_defconfig
Comment (fabiobaltieri) [inline boards/ct/ctcc_nrf52840/ctcc_nrf52840_defconfig:17]: this is deprecated, use the `gpio-as-nreset` dt property

**Comment by fabiobaltieri** [comment]
> Don't we need a sample to cover this board in CI? Also, should a maintainer be assigned for this board?

The weekly run should cover all boards automatically, the maintainer story for boards is unclear at this point unfortunately but as it stands we don't have areas for boards.

File: boards/ct/ctcc_nrf52840/board.yml
Comment (nordicjm) [inline boards/ct/ctcc_nrf52840/board.yml:2]: the board name should be ctcc without nrf52840, then it can be used as a base board and future soc variants added in future

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/ct/ctcc_nrf52840/board.yml
Comment (kowalewskijan) [reply boards/ct/ctcc_nrf52840/board.yml]: removed

File: boards/ct/ctcc_nrf52840/ctcc_nrf52840_defconfig
Comment (kowalewskijan) [reply boards/ct/ctcc_nrf52840/ctcc_nrf52840_defconfig]: thanks for pointing it out, removed

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by cvinayak** [comment]
> > Don't we need a sample to cover this board in CI? Also, should a maintainer be assigned for this board?
> 
> The weekly run should cover all boards automatically, the maintainer story for boards is unclear at this point unfortunately but as it stands we don't have areas for boards.

In the weekly, all samples are built for all boards? Wouldn't that fail if a board has no support features required by a sample?

**Comment by kowalewskijan** [comment]
@nordicjm @decsny - please see current changes

File: boards/ct/ctcc/doc/img/ctcc_nrf52840.png
Comment (nordicjm) [inline boards/ct/ctcc/doc/img/ctcc_nrf52840.png:1]: Would prefer this is converted to webp and put through https://tinypng.com/

File: boards/ct/ctcc/doc/index.rst
Comment (nordicjm) [inline boards/ct/ctcc/doc/index.rst:37]: no underscores, it would be `ctcc/nrf52840`

File: boards/ct/ctcc/doc/index.rst
Comment (nordicjm) [inline boards/ct/ctcc/doc/index.rst:42]: these are board targets (apply above comment also), board configuration does not exist anymore. Update all instances

File: boards/ct/ctcc/doc/index.rst
Comment (nordicjm) [inline boards/ct/ctcc/doc/index.rst:143]: commands need updating too

File: boards/ct/ctcc/doc/img/ctcc_nrf52840.png
Comment (kowalewskijan) [reply boards/ct/ctcc/doc/img/ctcc_nrf52840.png:1]: done

**Comment by kowalewskijan** [comment]
Fixed:
- [x] swapping `ctcc_nrf52840` to `ctcc/nrf52840` where it's referred as board target
- [x] converting png board image to webp

@nordicjm

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (nordicjm) [inline boards/ct/ctcc/ctcc_nrf52840.dts:59]: (non blocking) these sector sizes are not optimal, if you build MCUboot with debug logging it will print a message

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (nordicjm) [inline boards/ct/ctcc/ctcc_nrf52840.dts:20]: (non blocking) `zephyr,ieee802154 = &ieee802154;` could be added

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (nordicjm) [inline boards/ct/ctcc/ctcc_nrf52840.dts:17]: (non blocking) `zephyr,uart-mcumgr = &cdc_acm_uart;`, `zephyr,bt-mon-uart = &cdc_acm_uart;` and `zephyr,bt-c2h-uart = &cdc_acm_uart;` could be added

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (nordicjm) [inline boards/ct/ctcc/ctcc_nrf52840.dts:74]:
```
&gpio1 {
        status = "okay";
};
```
missing

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (nordicjm) [inline boards/ct/ctcc/ctcc_nrf52840.dts:75]:
```
&ieee802154 {
        status = "okay";
};
```
missing

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (kowalewskijan) [reply boards/ct/ctcc/ctcc_nrf52840.dts:74]: GPIO1 is not used on this board - none P1.n are connected to any interface. I guess we can skip that?

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (kowalewskijan) [reply boards/ct/ctcc/ctcc_nrf52840.dts:59]: Right, I can see with max dbg logs it won't fit into flash. However I'd leave it up to user to change mcuboot partition if needed.

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (nordicjm) [reply boards/ct/ctcc/ctcc_nrf52840.dts:74]: Sure, if all port 1 pins are disconnected. If they are present and can be connected but just aren't in the base file here then this should be present to allow users to use them in overlays.

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (nordicjm) [reply boards/ct/ctcc/ctcc_nrf52840.dts:59]: well the sizes here should be good defaults, if using the default MCUboot configuration then slot0 should be 2 sectors larger than slot1

Comment (kowalewskijan) [reply boards/ct/ctcc/ctcc_nrf52840.dts:59]: I got it - I'll change it to the partition layout from `nrf52840dongle` `fstab-debugger.dtsi`

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (kowalewskijan) [reply boards/ct/ctcc/ctcc_nrf52840.dts:24]: added

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (kowalewskijan) [reply boards/ct/ctcc/ctcc_nrf52840.dts:83]: added

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (kowalewskijan) [reply boards/ct/ctcc/ctcc_nrf52840.dts:20]: added, thanks

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (nordicjm) [inline boards/ct/ctcc/ctcc_nrf52840.dts:63]: not really sure what went on here?
Reduce boot by 0x2000 and make slot0 0x77000, keep slot1 at 0x75000

Comment (kowalewskijan) [reply boards/ct/ctcc/ctcc_nrf52840.dts:63]: I don't understand, how this will make enabled dbg logs from MCUboot fit into `mcuboot` partition? Having mcuboot partition with size `0x10000` and enabled logs `CONFIG_LOG_DEFAULT_LEVEL=4`  `CONFIG_MCUBOOT_LOG_LEVEL_DBG=y` `CONFIG_LOG_OVERRIDE_LEVEL=4` will results with `region `FLASH' overflowed by 6308 bytes`. Also, shouldn't images have the same size? Or am I misunderstanding something?

Comment (nordicjm) [reply boards/ct/ctcc/ctcc_nrf52840.dts:63]: Ah, if your MCUboot needs to be bigger then the increased size for that is fine. MCUboot's overhead sizes depend upon what mode you use, if you use swap using move (the default) which allows for image reverts then you need 1 sector to shift the image up by and an additional sector for the swap status data, this is why slot0 should be 2 sectors larger than slot1. If you pick these sizes, the upgrade will work, but slot1 will have 2 sectors you can never use, so you could take them off and put them in storage or take one from slot1 and add it to slot0

Comment (kowalewskijan) [reply boards/ct/ctcc/ctcc_nrf52840.dts]: Thanks for the explanation! I added 2 sectors to slot0, so now everything should be fine.

File: boards/ct/ctcc/ctcc_nrf52840.dts
Comment (nordicjm) [inline boards/ct/ctcc/ctcc_nrf52840.dts:63]: you need to take 2 sectors off from here, your partitions overlay

Comment (kowalewskijan) [reply boards/ct/ctcc/ctcc_nrf52840.dts]: Correct! I forgot about it

**Review by nordicjm — DISMISSED** [review]

**Comment by kowalewskijan** [comment]
@decsny @fabiobaltieri - PTAL

**Comment by kowalewskijan** [comment]
Added missing `watchdog0 = &wdt0;`

**Review by decsny — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**
